app/services/people/importer.rb

```diff
@@ -2,60 +2,89 @@
 
 module People
   class Importer
-    PROFILE_IDENTIFIERS_MAPPING = {
+    NOMIS_IDENTIFIER_MAPPING = {
       'police_national_computer' => :pnc_number,
       'prison_number' => :prison_number,
       'criminal_records_office' => :cro_number,
     }.freeze
 
-    attr_accessor :nomis_attributes
-
     def initialize(nomis_attributes)
-      self.nomis_attributes = nomis_attributes
+      @nomis_attributes = nomis_attributes
     end
 
     def call
       person = Person.find_or_initialize_by(nomis_prison_number: nomis_attributes[:prison_number])
+      person.assign_attributes(person_attributes)
+
       profile = person.latest_profile || person.profiles.build
-      profile.assign_attributes(attributes)
-      # return profile rather than person so that save does both profile and person if both are new
+      profile.assign_attributes(profile_attributes)
+
       profile
     end
 
   private
 
-    def attributes
+    attr_reader :nomis_attributes
+
+    def person_attributes
+      nomis_attributes.slice(:last_name, :date_of_birth).merge(
+        first_names: first_names,
+        gender: gender,
+        ethnicity: ethnicity,
+        police_national_computer: police_national_computer,
+        prison_number: prison_number,
+        criminal_records_office: criminal_records_office,
+        last_synced_with_nomis: last_synced_with_nomis,
+      )
+    end
+
+    def profile_attributes
       nomis_attributes.slice(:last_name, :date_of_birth, :aliases).merge(
         first_names: first_names,
-        gender_id: find_gender,
-        ethnicity_id: find_ethnicity,
+        gender: gender,
+        ethnicity: ethnicity,
         profile_identifiers: profile_identifiers,
         latest_nomis_booking_id: latest_nomis_booking_id,
+        last_synced_with_nomis: last_synced_with_nomis,
       )
     end
 
-    def find_gender
-      gender_param = nomis_attributes[:gender]
-      gender_param ? Gender.find_by(nomis_code: gender_param).id : nil
+    def gender
+      Gender.find_by(nomis_code: nomis_attributes[:gender])
     end
 
-    def find_ethnicity
-      ethnicity_param = nomis_attributes[:ethnicity]
-      ethnicity_param ? Ethnicity.find_by(title: ethnicity_param).id : nil
+    def ethnicity
+      Ethnicity.find_by(title: nomis_attributes[:ethnicity])
```

Comment:
just a sanity check here: we are finding the ethnicity by title, e.g. `Mixed: Any other background` as opposed to nomis code, e.g. `M9` ?? it seems a bit odd.

Reply:
Yeah, is a bit strange.